### PR TITLE
SUP-1067: Organization resource conversion to Framework

### PR DESCRIPTION
**Pull request by james2791**
## PR checklist:
- [x] `docs/` updated
- [x] tests added
- [x] an example added to `examples/` (useful to demo new field/resource)
- [x] `CHANGELOG.md` updated with pending release information

### CODE REVIEW

File: buildkite/provider.go [hunk 1]
```diff
@@ -135,11 +136,10 @@ func New(version string) provider.Provider {
 func Provider(version string) *schema.Provider {
 	provider := &schema.Provider{
 		ResourcesMap: map[string]*schema.Resource{
-			"buildkite_pipeline":              resourcePipeline(),
-			"buildkite_pipeline_schedule":     resourcePipelineSchedule(),
-			"buildkite_team":                  resourceTeam(),
-			"buildkite_team_member":           resourceTeamMember(),
-			"buildkite_organization_settings": resourceOrganizationSettings(),
```
Comment (jradtilbrook) [inline buildkite/provider.go:142]: This is still draft but one first comment: remember we will need to maintain this name `buildkite_organization_settings` as well until v1 or at least for a little while

File: buildkite/resource_organization_settings.go [hunk 2]
```diff
@@ -3,114 +3,262 @@ package buildkite
...
+type organizationResource struct {
+	client *Client
+}
+
+func newOrganizationResource() resource.Resource {
+	return &organizationResource{}
+}
+
+func (organizationResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_organization_settings"
```
Comment (mcncl) [inline buildkite/resource_organization_settings.go:32]: Given the `datasource` is just called `organization`, would we not do the same here?

Comment (jradtilbrook) [reply buildkite/resource_organization_settings.go:32]: Yeah thats right. The new resource built using the framework (this one) is `buildkite_organization` and we will leave the old one around using sdkv2 named `buildkite_organization_settings`. After some deprecation period, we can remove it. Lets say by version v0.21.0 - this will need documenting as well

File: docs/resources/organization_settings.md [hunk 3]
```diff
@@ -9,7 +9,7 @@ Note: The "Allowed API IP Addresses" feature must be enabled on your organizatio
 ## Example Usage
 
 ```hcl
-resource "buildkite_organization_settings" "test_settings" {
+resource "buildkite_organization_settings "test_settings" {
```
Comment (jradtilbrook) [inline docs/resources/organization_settings.md:12]: Mistakenly removed `"`?

File: buildkite/resource_organization_settings_test.go [hunk 4]
```diff
@@ -86,29 +86,12 @@ func TestAccOrganizationSettings_import(t *testing.T) {
 	})
 }
 
-func TestAccOrganizationSettings_disappears(t *testing.T) {
```
Comment (jradtilbrook) [inline buildkite/resource_organization_settings_test.go:89]: Why is this removed? As with most of these migrations, we shouldn't need to change the acceptance tests

Comment (jradtilbrook) [reply buildkite/resource_organization_settings_test.go:89]: We also need a test against the new resource name buildkite_organization

**Review by jradtilbrook — CHANGES_REQUESTED** [review]
Its looking good 👍
Im confused about why the getOrganization function is getting called everwhere. FWIW you can get the org ID from client.organizationID

File: buildkite/resource_organization_settings_test.go (hunk 4 above)
Comment (mcncl) [reply buildkite/resource_organization_settings_test.go:89]: Yeah, this can technically be removed if you’re checking for the removal on the Create test, but given that only tests that the resource is added you’ll need to check it has been removed too.

File: docs/resources/organization.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Resource: organization
```
Comment (jradtilbrook) [inline docs/resources/organization.md:1]: I think we should call out in the docs for `organization_settings` that it is deprecated as well

Comment (jradtilbrook) [reply docs/resources/organization.md:1]: And you can also add a `Deprecated` field to the schema of the SDKv2 implementation so it will output on the log when running terraform

File: buildkite/resource_organization.go [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+		return
+	}
+
+	if response.Organization.Id == "" {
+		resp.Diagnostics.AddError(
+			"Organization not found",
+			fmt.Sprintf("Organization not found: %s", err.Error()),
+		)
+		return
+	}
```
Comment (jradtilbrook) [inline buildkite/resource_organization.go:92]: These errors will surface from the request on line 98 itself so to save an extra request slowing down the provider we can remove these. There's a few below in the other functions as well

**Review by jradtilbrook — CHANGES_REQUESTED** [review]

File: buildkite/resource_organization_settings.go [hunk 7]
```diff
@@ -9,6 +9,9 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+var deprecationMessage = `This resource has been deprecated in favour of the newer buildkite_organization resource. 
+Please visit the provider's documentation at https://registry.terraform.io/providers/buildkite/buildkite/latest/docs fo more details.`
```
Comment (mcncl) [inline buildkite/resource_organization_settings.go:13]: "fo" 😉

**Review by mcncl — APPROVED** [review]

**Review by jradtilbrook — APPROVED** [review]

**Review by mcncl — APPROVED** [review]
🚀

**Comment by james2791** [comment]
🎆
